### PR TITLE
Add details for the collect_geometries function

### PR DESCRIPTION
**Pull request by DelazJ**
Ticket(s): #4173 (follow-up #4271 )

### CODE REVIEW

File: source/docs/user_manual/style_library/symbol_selector.rst [hunk 1]
```diff
@@ -349,6 +349,15 @@ Some examples:
...
            ) 
 
+  -- Create a nice radial effect of points surrounding the central feature
+  -- point when used as a MultiPoint geometry generator
+  collect_geometries(
+    array_foreach(
+      generate_series( 0, 330, 30 ),
+        project( $geometry, .2, radians( @element ) )
+    )
+  )
```
Comment (DelazJ) [inline source/docs/user_manual/style_library/symbol_selector.rst]: @nyalldawson I fail to get anything in the map canvas using this code I copied from your PR. I'm trying to use it as a symbol layer. Any idea what is missing or did I misunderstand the use case? my build is at https://github.com/qgis/QGIS/commit/70fe6c8c23ee3bc9df2170a5c1747bb141516ea3 if relevant

Comment (nyalldawson) [reply source/docs/user_manual/style_library/symbol_selector.rst]: What's your map units?

Comment (DelazJ) [reply source/docs/user_manual/style_library/symbol_selector.rst]: Using documentation data sample, I tried with epsg 4326, 3857, 2964 so I guess, degrees, meters and feet

Comment (DelazJ) [reply source/docs/user_manual/style_library/symbol_selector.rst]: @nyalldawson I tried with another project in meters and it works perfectly. Thanks and sorry for the noise.

**Pull request merged**